pkg/snmp/traps/config.go

```diff
@@ -60,12 +82,90 @@ func (c *Config) Addr() string {
 	return fmt.Sprintf("%s:%d", c.BindHost, c.Port)
 }
 
-// BuildV2Params returns a valid GoSNMP SNMPv2 params structure from configuration.
-func (c *Config) BuildV2Params() *gosnmp.GoSNMP {
-	return &gosnmp.GoSNMP{
-		Port:      c.Port,
-		Transport: "udp",
-		Version:   gosnmp.Version2c,
-		Logger:    gosnmp.NewLogger(&trapLogger{}),
+func (c *Config) getAgentHostname() string {
+	hostname, err := util.GetHostname(context.TODO())
+	if err != nil {
+		// this scenario is not likely to happen since
+		// the agent cannot start without a hostname
+		hostname = "unknown-datadog-agent"
 	}
+	return hostname
+}
+
+// BuildAuthoritativeEngineID creates a engineID byte-array from the OS hostname
+func (c *Config) BuildAuthoritativeEngineID() [28]byte {
+	engineID := [28]byte{}
+	// First byte is always 0x80
+	// Next four bytes are the Private Enterprise Number (set to an invalid value here)
+	copy(engineID[:5], []byte{0x80, 0xff, 0xff, 0xff, 0xff})
+	copy(engineID[5:], []byte(c.getAgentHostname()))
```

Comment:
Shall we also include namespace as part of the engineID ?

Reply:
No need to. We just want the engineID to be unique on a given subdomain (and even then, having overlapping engineID for sending traps isn't a big deal). By definition when `namespace` is used to differentiate between hosts, it's not a problem to share the engineID.

What's quite important is to have something that's consistent across agent restarts